packages/abstractions/src/requestInformation.ts

```diff
@@ -252,6 +253,8 @@ export class RequestInformation implements RequestInformationSetContent {
               writer.writeTimeOnlyValue(undefined, value as any as TimeOnly);
             } else if (value instanceof Duration) {
               writer.writeDurationValue(undefined, value as any as Duration);
+            } else if (value instanceof Guid) {
```

Comment:
This if statement is for the request body, it's good to fix the issue here as well as people will eventually run into it, but it's not going to address the initial issue.

Reply:
@illunix 
I think there are 2 possible ways to convert a Guild value in path parameters to a string:

# Inside `RequestInformation.URL`

```typescript
public get URL(): string {
  const rawUrl = this.pathParameters[
    RequestInformation.raw_url_key
  ] as string;
  if (this.uri) {
    ...
  } else {
    const data = {} as { [key: string]: unknown };
    for (const key in this.queryParameters) {
      if (this.queryParameters[key] !== null && this.queryParameters[key] !== undefined) {
        data[key] = this.queryParameters[key];
      }
    }
    for (const key in this.pathParameters) {
      if (this.pathParameters[key]) {
        data[key] = this.pathParameters[key]; // <- here
      }
    }
    return StdUriTemplate.expand(this.urlTemplate, data);
  }
}

```

# Inside `apiClientProxifier`

```ts
export function apiClientProxifier<T extends object>(
  requestAdapter: RequestAdapter,
  pathParameters: Record<string, unknown>,
  navigationMetadata?: Record<string, NavigationMetadata>,
  requestsMetadata?: RequestsMetadata,
): T {
  if (!requestAdapter) throw new Error("requestAdapter cannot be undefined");
  if (!pathParameters) throw new Error("pathParameters cannot be undefined");
  return new Proxy({} as T, {
    get(target, property) {
      ...
      if (navigationMetadata) {
        const navigationCandidate = navigationMetadata[name];
        if (navigationCandidate) {
          ...
          return (...argArray: any[]) => {
            // navigation method like indexers or multiple path parameters
            const downWardPathParameters = getPathParameters(pathParameters);
            if (
              navigationCandidate.pathParametersMappings &&
              navigationCandidate.pathParametersMappings.length > 0
            ) {
              for (let i = 0; i < argArray.length; i++) {
                const element = argArray[i];
                downWardPathParameters[
                  navigationCandidate.pathParametersMappings[i]
                ] = element; // <- here
              }
            }
            return apiClientProxifier(
              requestAdapter,
              downWardPathParameters,
              navigationCandidate.navigationMetadata,
              navigationCandidate.requestsMetadata,
            );
          };
        }
        throw new Error(
          `couldn't find navigation property ${name} data: ${JSON.stringify(
            navigationMetadata,
          )}`,
        );
      }
    },
  });
}
```

I hope that this will help you. Thanks